src/sagemaker/utils.py

```diff
@@ -346,16 +346,21 @@ def download_folder(bucket_name, prefix, target, sagemaker_session):
 
     # Try to download the prefix as an object first, in case it is a file and not a 'directory'.
     # Do this first, in case the object has broader permissions than the bucket.
-    try:
-        s3.Object(bucket_name, prefix).download_file(os.path.join(target, os.path.basename(prefix)))
-        return
-    except botocore.exceptions.ClientError as e:
-        if e.response["Error"]["Code"] == "404" and e.response["Error"]["Message"] == "Not Found":
-            # S3 also throws this error if the object is a folder,
-            # so assume that is the case here, and then raise for an actual 404 later.
-            _download_files_under_prefix(bucket_name, prefix, target, s3)
-        else:
-            raise
+    if not prefix.endswith("/"):
+        try:
+            file_destination = os.path.join(target, os.path.basename(prefix))
+            s3.Object(bucket_name, prefix).download_file(file_destination)
+            return
+        except botocore.exceptions.ClientError as e:
+            err_info = e.response["Error"]
+            if err_info["Code"] == "404" and err_info["Message"] == "Not Found":
+                # S3 also throws this error if the object is a folder,
+                # so assume that is the case here, and then raise for an actual 404 later.
```

Comment:
If there's an actual 404, where would the error be raised?

Reply:
in `_download_files_under_prefix`: https://github.com/aws/sagemaker-python-sdk/blob/8e06a27ea2d034ac07050774e379a8d7fc37f516/src/sagemaker/utils.py#L391